sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala

```diff
@@ -184,7 +188,110 @@ private[cosmos] object SparkBridgeImplementationInternal {
       .setIoThreadPriority(config, ioThreadPriority)
   }
 
-  def setUserAgentWithSnapshotInsteadOfBeta() = {
-    HttpConstants.Versions.useSnapshotInsteadOfBeta();
+  def setUserAgentWithSnapshotInsteadOfBeta(): Unit = {
+    HttpConstants.Versions.useSnapshotInsteadOfBeta()
+  }
+
+  def createChangeFeedOffsetFromSpark2
+  (
+    client: CosmosAsyncClient,
+    databaseResourceId: String,
```

Comment:
is the databaseResourceId here == databaseId/databaseName?

Reply:
Backend handles rid and id the same - passing in _rid works. I have tested it. Reading teh database with rid to get the Id is the only way how you can do the mapping when id is unknown. And in Spark 2.4 state we don't know the id - just the _rid.

Reply:
ah gotcha, did not aware that _rid and id will both works